rest/route/admin_routes.go

```diff
@@ -217,3 +218,53 @@ func (h *flagsPostHandler) Execute(ctx context.Context, sc data.Connector) (Resp
 		Result: []model.Model{&h.Flags},
 	}, nil
 }
+
+// this manages the /admin/restart route, which restarts failed tasks
+func getRestartRouteManager(route string, version int) *RouteManager {
+	rh := &restartHandler{}
+	restartHandler := MethodHandler{
+		PrefetchFunctions: []PrefetchFunc{PrefetchUser},
+		Authenticator:     &SuperUserAuthenticator{},
+		RequestHandler:    rh.Handler(),
+		MethodType:        evergreen.MethodPost,
+	}
+
+	restartRoute := RouteManager{
+		Route:   route,
+		Methods: []MethodHandler{restartHandler},
+		Version: version,
+	}
+	return &restartRoute
+}
+
+type restartHandler struct {
+	StartTime time.Time `json:"start_time"`
+	EndTime   time.Time `json:"end_time"`
+	DryRun    bool      `json:"dry_run"`
+}
+
+func (h *restartHandler) Handler() RequestHandler {
+	return &restartHandler{}
+}
+
+func (h *restartHandler) ParseAndValidate(ctx context.Context, r *http.Request) error {
+	if err := util.ReadJSONInto(r.Body, h); err != nil {
```

Comment:
do we need to make sure we defer close the body?

Reply:
done